python/ngen_conf/src/ngen/config/hydrofabric.py

```diff
@@ -0,0 +1,74 @@
+
+from typing import List, Literal
+from pydantic import BaseModel, validator
+from geojson_pydantic import Feature, Point,  MultiPolygon
+
+def check_in_accepted(val, accepted):
+    if val not in accepted:
+        raise ValueError(f'Not an acceptable option. Options are {accepted}')
+    
+class NexusFeatureProperty(BaseModel):
+    toid: str
+
+    @validator('toid')
```

Comment:
I should have mentioned this earlier, so my apologies for not, the `toid` prefix (i.e. `cat-`) has not yet stabilized in the hydrofabric. The older hydrofabric's use `cat-`, but newer versions are using `wb-` and I forget, but I seen other prefixes too. I wish I has a good answer for why...

I think the best bet here is to use a regular expression to validate the `toid` field. Luckily, the all do follow the pattern `{{ some-prefix }}-{{ numbers }}`, so you should be able to use the regular expression `"-\d+$"`. In practice that will look like:

```python
import re

# match str that ends with a '-' followed by digits
_TOID_REGEX_STR = "-\d+$"
_TOID_REGEX = re.compile(_TOID_REGEX_STR)

class NexusFeatureProperty(BaseModel):
    toid: str

    @validator('toid')
    def _check_prefix(cls, value: str) -> str:
        if _TOID_REGEX.search(value) is None:
            raise ValueError(f"Invalid 'toid' property value: {value!r}")
        return value

```

The same applies `CatchmentFeatureProperty`, the `toid` prefix is now either `nex-` or `tnx-` (terminal nexus), but just to be on the safe side, I would use the regex for this too. So really you could reuse the validator if you want like:


```python
def _validate_prefix(value: str) -> str:
    if _TOID_REGEX.search(value) is None:
        raise ValueError(f"Invalid 'toid' property value: {value!r}")
    return value
```

Then in the models just do:

```python
class NexusFeatureProperty(BaseModel):
    toid: str

    _validate_prefix = validator("toid", allow_reuse=True)(_validate_prefix)
```

Sorry, that is a lot. There is a fair amount of head scratching when it comes to ngen configuration files.

Reply:
This all makes sense. I added the condition that the characters before the hyphen can't contain a number. I think it's possible to require one of several letter combinations ("cat" or "wb") if we want.